spring-boot-project/spring-boot-docs/src/main/asciidoc/deployment.adoc

```diff
@@ -35,17 +35,23 @@ Once you have unpacked the jar file, you can also get an extra boost to startup
 
 More efficient container images can also be created by copying the dependencies to the image as a separate layer from the application classes and resources (which normally change more frequently).
 There is more than one way to achieve this layer separation.
-For example, using a `Dockerfile` you could express it in this form (assuming the jar is already unpacked at `target/dependency`):
+For example, using a `Dockerfile` you could express it in this form:
 
 [indent=0]
 ----
-	FROM openjdk:8-jdk-alpine
-	VOLUME /tmp
-	ARG DEPENDENCY=target/dependency
-	COPY ${DEPENDENCY}/BOOT-INF/lib /app/lib
-	COPY ${DEPENDENCY}/META-INF /app/META-INF
-	COPY ${DEPENDENCY}/BOOT-INF/classes /app
-	ENTRYPOINT ["java","-cp","app:app/lib/*","com.example.MyApplication"]
+    FROM openjdk:8-jdk-alpine AS builder
+    WORKDIR target/dependency
+    COPY ./target/*.jar .
```

Comment:
There might be more than one jar in this directory. The regular build and the fat jar output. A build may also have other classified artifacts and we can't really know in advance which one is going to be the fatjar.

Reply:
Ok I was focused on the user flow experience, and try to resolve the forgetting of unzip the jar, got your point maybe I should remove this section

Reply:
Or maybe we can use a build arg with the name of the fatjar & add a command example for the docker build, wdyt?

Reply:
`target` is a usual location for Maven but I like the idea of the argument. Can we improve this so that we give the (full) path to the fat jar and then it does everything else without assuming a directory?

Reply:
yes of course, I'll update in that way 😉